lxd/storage/drivers/driver_alletra_wsapi1.go

```diff
@@ -11,3 +40,237 @@ func newAlletraClient(driver *alletra) *alletraClient {
 		driver: driver,
 	}
 }
+
+// hpeError represents an error response from the HPE Storage API.
+type hpeError struct {
+	Code           int    `json:"code"`
+	Desc           string `json:"desc"`
+	HTTPStatusCode int
+}
+
+// Error implements the error interface for hpeError.
+func (p *hpeError) Error() string {
+	if p == nil {
+		return ""
+	}
+
+	return fmt.Sprintf("HTTP Error Code: %d. Alletra WSAPI Error Code: %d. Alletra WSAPI Description: %s", p.HTTPStatusCode, p.Code, p.Desc)
+}
+
+// isHpeErrorOf checks if the error is of type hpeError and matches the status code.
+func isHpeErrorOf(err error, statusCode int, substrings ...string) bool {
+	perr, ok := err.(*hpeError)
+	if !ok {
+		return false
+	}
+
+	if perr.Code != statusCode {
+		return false
+	}
+
+	if len(substrings) == 0 {
+		return true
+	}
+
+	errMsg := strings.ToLower(perr.Desc)
+	for _, substring := range substrings {
+		if strings.Contains(errMsg, strings.ToLower(substring)) {
+			return true
+		}
+	}
+
+	return false
+}
+
+// hpeIsNotFoundError returns true if the error is of type hpeError, its status code is 400 (bad request),
+// and the error message contains a substring indicating the resource was not found.
+func isHpeErrorNotFound(err error) bool {
```

Comment:
When we have the new `clients` package, this can be moved inside it, and then used to convert HPE specific not found errors to `api.StatusErrorf(http.StatusNotFound)` errors for use in the driver itself.